### PR TITLE
Support default setting overrides on different languages

### LINKED ISSUE #1328: Support default settings by language
Different languages work best with different configurations of Yomitan. We should consider suggesting default setting changes for certain languages and have them configured somewhere

### PR DESCRIPTION
**Pull request by jamesmaa**
Fixes #1328 

- [x] : Use a modal instead of `confirm`
- [x] : Applying overrides updates UI elements in settings page
- [x] : Schema validation for overrides
- [x] : Valid setting validation: auto caught by settings-controller
- [x] : Tests: not doing 

image

### CODE REVIEW

File: ext/js/pages/settings/languages-controller.js [hunk 1]
```diff
@@ -25,17 +25,25 @@ export class LanguagesController {
...
+        this._lastSelectedLanguage = '';
     }
 
     /** */
     async prepare() {
         const languages = await this._settingsController.application.api.getLanguageSummaries();
         languages.sort((a, b) => a.name.localeCompare(b.name, 'en'));
-        this._fillSelect(languages);
+        const languageSelect = this._fillSelect(languages);
+        languageSelect.addEventListener(
```
Comment (jamesmaa) [inline ext/js/pages/settings/languages-controller.js:37]: This is actually kind of brittle. To prompt users whether they'd like to update the current profile settings upon changing the language selection dropdown is falsely triggered by the following scenarios:

- When I change profiles which update the language selected
- When I load the page and the `prepare()` statement runs, changing the value (this is guarded via a clause)

If someone knows how to guard against the first condition OR know a better place to hook this logic, that would be great

**Review by Kuuuube — CHANGES_REQUESTED** [review]

**Comment by jamesmaa** [comment]
comments addressed. current look 
image

image

**Review by Kuuuube — APPROVED** [review]
I think this is fine to merge but I feel like we should probably add overrides to turn off the scan whole word stuff when selecting japanese, chinese, cantonese, and korean.

**Comment by MarvNC** [comment]
msedge_Support_default_setting_overrides_on_different_lan_2024-10-26_03-27-42

Playwright breakage seems to have been caused by this PR

**Pull request merged**